edx_django_utils/db/read_replica.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tools for making queries read from the read-replica database, rather than from the writer database.
+
+Settings:
+    EDX_READ_REPLICA_DB_NAME: The name of the read-replica in the DATABASES django setting.
+        Defaults to "read_replica".
+    EDX_WRITER_DB_NAME: The name of the writer in the DATABASES django setting.
+        Defaults to "writer".
+"""
+from contextlib import contextmanager
+from functools import partial
+import threading
+
+from django.conf import settings
+
+
+READ_REPLICA_NAME = getattr(settings, "EDX_READ_REPLICA_DB_NAME", "read_replica")
+WRITER_NAME = getattr(settings, "EDX_WRITER_DB_NAME", "default")
+
+READ_REPLICA_OR_DEFAULT = (
+    READ_REPLICA_NAME if READ_REPLICA_NAME in settings.DATABASES else WRITER_NAME
+)
```

Comment:
It's safe to do this because Django settings are lazy, right?

Reply:
Are you thinking about the fact that the database router is going to be referencing this file? The router will be named via a string, so it should be ok (it'll be loaded later)

Reply:
I was mostly thinking that I didn't know if Django's settings were always properly initialized here because these imports run at import, but I think that was only a problem on much earlier versions of Django (but didn't remember for sure). I think it's fine these days, but I just wanted to double check.